de.metas.contracts/src/main/java/de/metas/contracts/impl/FlatrateDAO.java

```diff
@@ -843,4 +841,21 @@ public I_C_Flatrate_Term retrieveAncestorFlatrateTerm(@NonNull final I_C_Flatrat
 				.create()
 				.firstOnly(I_C_Flatrate_Term.class);
 	}
+
+	@Override
+	public List<I_C_Invoice> retrieveInvoicesForCurrentContract(@NonNull final I_C_Flatrate_Term contract)
+	{
+		final Properties ctx = InterfaceWrapperHelper.getCtx(contract);
+		final String trxName = InterfaceWrapperHelper.getTrxName(contract);
+		final List<I_C_Invoice> currentFlatRateTermInvoices = new ArrayList<>();
+		final List<I_C_Invoice_Candidate> icsForCurrentTerm = Services.get(IInvoiceCandDAO.class).fetchInvoiceCandidates(ctx, I_C_Flatrate_Term.Table_Name, contract.getC_Flatrate_Term_ID(), trxName);
+		icsForCurrentTerm.forEach(ic -> {
+			final List<I_C_InvoiceLine> iclList = Services.get(IInvoiceCandDAO.class).retrieveIlForIc(ic);
+			iclList.stream()
+					.filter(il -> !currentFlatRateTermInvoices.contains(il.getC_Invoice()))
```

Comment:
hi, i just researched a bit and added (to the `master` branch)
`StreamUtils.distinctByKey()` which returns a filter predicate..with this, you could write something like
```java
iclList.stream()
  .filter(StreamUtils.distinctByKey(I_C_InvoiceLine::getC_Invoice_ID))
  .map(il -> il.getC_Invoice())
  .collect(Collectors.asList());
```
i think it's more readable, and we don't have to assume that I_C_Invoice.equals works our way

Reply:
Other thing: i think codacy has a hint for a quick-fix for you:
"Unnecessary use of fully qualified name 'InterfaceWrapperHelper.save' due to existing static import 'org.adempiere.model.InterfaceWrapperHelper.save'"